JavadocStyleGuide.md

```diff
@@ -0,0 +1,223 @@
+# Gradle Javadoc Style Guide
+
+## 1.1 Formatting
+
+### 1.1.1 General form
+
+The basic formatting of Javadoc blocks is as seen in this example:
+
+```java
+/**
+ * Returns an Image object that can then be painted on the screen.
+ * The url argument must specify an absolute <a href="#{@link}">{@link URL}</a>. The name
```

Comment:
```suggestion
 * Returns an Image object that can then be painted on the screen.
 * <p>
 * The url argument must specify an absolute <a href="#{@link}">{@link URL}</a>. The name
```

I think we should always have one short sentence describing the "Thing", and then a newline. In this example, that makes total sense as well.

Reply:
IDEA seems to disagree: 
<img width="683" alt="image" src="https://github.com/gradle/gradle/assets/423186/ecf6bef7-c7c0-4988-855a-c24889028c80">

Aka it doesn't see that there is a paragraph after the summary.

Reply:
I thought no paragraph symbol was needed as well - and perhaps it isn't for compiled HTML javadoc - but if we can't get IDEA to figure out how to add paragraphs when Javadoc is written with only blank lines, we should require the `<p>`.  Viewing in the IDE is an important use case.

Reply:
Style guide has been updated, IntelliJ IDEA sees <p>, <br>, and * blank line as the same thing.